### DISCUSSION TITLE
[QUESTION]:  ERROR [2/6] RUN apk update && apk add python make g++

### DISCUSSION THREAD

**Discussion by dzx2018** [Netflix/conductor#2490] [Q&A] [answered]
=> ERROR [2/6] RUN apk update && apk add python make g++                                                                                                              1.8s
------                                                                                                                                                                      
 > [2/6] RUN apk update && apk add python make g++:                                                                                                                         
#5 0.427 fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/aarch64/APKINDEX.tar.gz                                                                                      
#5 1.172 fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/aarch64/APKINDEX.tar.gz
#5 1.172 ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.11/main: Protocol error
#5 1.172 WARNING: Ignoring http://dl-cdn.alpinelinux.org/alpine/v3.11/main: No such file or directory
#5 1.756 ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.11/community: Protocol error
#5 1.756 WARNING: Ignoring http://dl-cdn.alpinelinux.org/alpine/v3.11/community: No such file or directory
#5 1.757 2 errors; 16 distinct packages available
------
executor failed running [/bin/sh -c apk update && apk add python make g++]: exit code: 2

**Comment by aravindanr** [comment]
Can you be more specific with the question? This does not seem to be related to Conductor.

**Reply by javiercanillas** [reply]
Just hit the same issue: 
```bash
Building conductor-ui
[+] Building 4.0s (7/11)
 => [internal] load build definition from Dockerfile                                                                                                                                                                                          0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                             0.0s
 => => transferring context: 34B                                                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/node:14-alpine                                                                                                                                                                             2.2s
 => [auth] library/node:pull token for registry-1.docker.io                                                                                                                                                                                   0.0s
 => CACHED [1/6] FROM docker.io/library/node:14-alpine@sha256:c346198378f78f8611254dce222e7e6635804e41e5203d1825321edd6c59dca1                                                                                                                0.0s
 => [internal] load build context                                                                                                                                                                                                             0.0s
 => => transferring context: 5.39kB                                                                                                                                                                                                           0.0s
 => ERROR [2/6] RUN apk update && apk add python make g++                                                                                                                                                                                     1.7s
------
 > [2/6] RUN apk update && apk add python make g++:
#6 0.306 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
#6 0.894 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
#6 1.458 v3.14.2-140-gf8961fd93a [https://dl-cdn.alpinelinux.org/alpine/v3.14/main]
#6 1.458 v3.14.2-140-gf8961fd93a [https://dl-cdn.alpinelinux.org/alpine/v3.14/community]
#6 1.458 OK: 14945 distinct packages available
#6 1.691 ERROR: unable to select packages:
#6 1.692   python (no such package):
#6 1.727     required by: world[python]
------
executor failed running [/bin/sh -c apk update && apk add python make g++]: exit code: 1
ERROR: Service 'conductor-ui' failed to build : Build failed
```

I thought it was related to the python command itself, but it seems `python` alias is pointing to version 3.8.8, so it's already python3 ( as I read somewhere that it should be):
```bash
$ python --version
Python 3.8.8
```

Versions:
* Docker version is: 4.1.1 (69879),
* Engine: 20.10.8
* Compose: 1.29.2

Plus, I'm running on Mac OSX Monterrey, version 12.0.1.

Any guideness on how to troubleshoot this to discover the internal issue would be welcomed, I'm quite new to docker universe.

Plus, executing the suggestion command ends up with an error telling `apk` is unknown:
```bash
$ apk update && apk add python make g++:
-bash: apk: command not found
$ /bin/sh -c apk update && apk add python make g++
update: apk: command not found
```

**Comment by javiercanillas** [comment, marked as answer]
Ok, found a workaround for this.. on the file `docker/ui/Dockerfile` in line 9, we need to change:
```bash
RUN apk update && apk add python make g++
```
to
```bash
RUN apk update && apk add --no-cache python3 py3-pip  make g++
```

**Comment by peterlau** [comment]
@javiercanillas  Thanks for the workaround. Will prioritize patch.

**Reply by peterlau** [reply]
Patch released in 3.3.5
